include/linux/blkdev_compat.h

```diff
@@ -37,36 +37,26 @@ typedef unsigned __bitwise__ fmode_t;
 #endif /* HAVE_FMODE_T */
 
 /*
- * 2.6.36 API change,
+ * 4.7 - 4.x API,
+ * The blk_queue_write_cache() interface has replaced blk_queue_flush()
+ * interface.  However, the new interface is GPL-only thus we implement
+ * our own trivial wrapper when the GPL-only version is detected.
+ *
+ * 2.6.36 - 4.6 API,
  * The blk_queue_flush() interface has replaced blk_queue_ordered()
  * interface.  However, while the old interface was available to all the
  * new one is GPL-only.   Thus if the GPL-only version is detected we
- * implement our own trivial helper compatibility funcion.   The hope is
- * that long term this function will be opened up.
- */
-#if defined(HAVE_BLK_QUEUE_FLUSH) && defined(HAVE_BLK_QUEUE_FLUSH_GPL_ONLY)
-#define	blk_queue_flush __blk_queue_flush
-static inline void
-__blk_queue_flush(struct request_queue *q, unsigned int flags)
-{
-	q->flush_flags = flags & (REQ_FLUSH | REQ_FUA);
-}
-#endif /* HAVE_BLK_QUEUE_FLUSH && HAVE_BLK_QUEUE_FLUSH_GPL_ONLY */
-
-/*
- * 4.7 API change,
- * The blk_queue_write_cache() interface has replaced blk_queue_flush()
- * interface.  However, while the new interface is GPL-only. Thus if the
- * GPL-only version is detected we implement our own trivial helper
- * compatibility funcion.
+ * implement our own trivial helper.
+ *
+ * 2.6.x - 2.6.35
+ * Legacy blk_queue_ordered() interface.
  */
-#if defined(HAVE_BLK_QUEUE_WRITE_CACHE) && \
-	defined(HAVE_BLK_QUEUE_WRITE_CACHE_GPL_ONLY)
-#define	blk_queue_write_cache __blk_queue_write_cache
 static inline void
-__blk_queue_write_cache(struct request_queue *q, bool wc, bool fua)
+blk_queue_set_write_cache(struct request_queue *q, bool wc, bool fua)
 {
 	spin_lock_irq(q->queue_lock);
```

Comment:
The spinlock does exist all the way back to 2.6.32 but your right it is a small functional change, which I'm pretty sure is safe.  But I wouldn't object to just moving it back to the one case.